### PR TITLE
Support for -gdwarf-5 in flang2

### PR DESCRIPTION
**Pull request by alokkrsharma**
Summary:
  flang2 doesnt produce dwarf version 5 for -gdwarf-5.

  Currently when -gdwarf-5 is passed to flang (driver), that doesnt
pass any corresponding (-x 120) option as it passes in case of
-gdwarf-2/3. flang2 binary picks -gdwarf-4 by default.
   # flang -gdwarf-5 test.f90
   # llvm-dwarfdump a.out | grep version
0x00000000: Compile Unit: length = 0x0000008e version = 0x0004

  Now flang2 will identify gdwarf-4 and gdwarf-5 by -x 120 0x1000000
and -x 120 0x2000000, flang(driver) need to pass these option to flang

  flang-driver changes will be pushed separately. After that -gdwarf-5
will work from glang driver.

Testing:
  - check-llvm
  - check-debuginfo

Change-Id: Ibb70438f0a024bb7a6425d8b022c9353d138fa98

### CODE REVIEW

File: tools/flang2/flang2exe/ll_structure.cpp
Comment (schweitzpgi) [inline tools/flang2/flang2exe/ll_structure.cpp]: Can you make this a simple `else` clause? And add a comment that DWARF 4 is the default? (This will fix the FIXME. Yay.)

**Comment by schweitzpgi** [comment]
@jiel-nv Should we have some sort of test for this, to make sure DWARF-5 is really being output?

File: tools/flang2/flang2exe/ll_structure.cpp
Comment (alokkrsharma) [reply tools/flang2/flang2exe/ll_structure.cpp]: Thanks a lot for your comment. I shall be updating with your suggestion.

**Comment by jiel-nv** [comment]
> @jiel-nv Should we have some sort of test for this, to make sure DWARF-5 is really being output?

Yes, I think so. I will work it out.

**Comment by nextsilicon-itay-bookstein** [comment]
I noticed that the other bits controlling the DWARF version are used for instance in `tools/flang2/flang2exe/main.cpp` in the definitions of `IS_DWARF` and `IS_COFFDBG`. Should they be modified to include these two new possibilities as well?

**Comment by alokkrsharma** [comment]
> 
> 
> I noticed that the other bits controlling the DWARF version are used for instance in `tools/flang2/flang2exe/main.cpp` in the definitions of `IS_DWARF` and `IS_COFFDBG`. Should they be modified to include these two new possibilities as well?

Thanks for bringing this to notice. I also noticed that these defines are unused. Can you are someone else suggest whether these defines need update for newer bits ?

**Comment by nextsilicon-itay-bookstein** [comment]
If it were up to me I'd delete the dead code rather than update it, but I defer to the project's maintainers in deciding this - I don't know the history/context behind that piece of dead code :)

**Review by gklimowicz — APPROVED** [review]
I tested this PR along with #888 and #921, and the Flang-driver PR#92 and PR#93. These pass tests on OpenPOWER. The entire collection looks good to me.

**Review by kiranchandramohan — APPROVED** [review]
LGTM.

**Review by bryanpkc — APPROVED** [review]
LGTM.

**Pull request merged**
